### PR TITLE
#160577476 Users Verification after signup

### PR DESCRIPTION
**Pull request by ibetween**
### What does this PR do?

-  Verify user after signup
### Description of Task to be completed?
- The users can verify their account after signup.
- A link is sent to the email. After the user clicks the link they will be able to activate their account.
 `Note: Before clicking the link, the account is inactive`
### How should this be manually tested?
- Clone [ah-shakas](https://github.com/andela/ah-shakas) repo `$ git clone https://github.com/andela/ah-shakas.git`
- export db credentials. please note that there is no spaces before and after `=`

```
      export DB_NAME="name of your database"
      export DB_PASSWORD="passord to your database user"
      export DB_HOST="your local server eg (127.0.0.1)"
      export DB_USER="your database username eg (postgres)"
```
_Note- if you have no password set for your database user leave the password field blank i.e `export DB_PASSWORD=""`_
- On the project's root directory run `python manage.py runserver`
- send a `POST` request to the server http://127.0.0.1:8000/users/ whith the following in the body.
```
{
  "user":{
    "email":"email@domain.com",
    "username":"simple username",
    "password":"AlphaNumericPassword"
  }
}
```
- Done! 🙂
### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/160577476

### CODE REVIEW

**Comment by koechkevin** [comment]
please resolve merge conflicts

**Comment by ibetween** [comment]
> please resolve merge conflicts

Thanks. Will resolve ASAP

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -1,33 +1,61 @@
...
+            'name': user_name
+        })
+        send_mail(
+                    'Verify your email',
+                    'Please verify your account.',
+                    'no-reply@authors-heaven.com',
+                    [user_email],
+                    html_message = body,
+                    fail_silently=False,
+                )
```
Comment (FawazFarid) [inline authors/apps/authentication/views.py]: this indentation, seems a bit off, let's do it this way:
```
send_mail(
    'Verify your email',
    'Please verify your account.',
    'no-reply@authors-heaven.com',
    [user_email],
    html_message = body,
    fail_silently=False,
)
```

File: authors/apps/authentication/views.py [hunk 2]
```diff
@@ -1,33 +1,61 @@
...
+        return_message = {"Message":"Thank you for registering at Authors heaven. To start using authors heaven, go to your email and click the confirmation link which we haves sent to you :D"}
```
Comment (FawazFarid) [inline authors/apps/authentication/views.py]: this line is too long

File: authors/settings.py [hunk 3]
```diff
@@ -152,4 +152,19 @@
...
+#send grid credentials
+EMAIL_HOST = os.getenv('EMAIL_HOST')
+EMAIL_HOST_USER = os.getenv('EMAIL_HOST_USER')
+EMAIL_HOST_PASSWORD = os.getenv('EMAIL_HOST_PASSWORD')
+EMAIL_PORT = 587
+EMAIL_USE_TLS = True
+
+
+# app default domain
+DEFAULT_DOMAIN = 'http://127.0.0.1:8000'
```
Comment (FawazFarid) [inline authors/settings.py]: what if we push this to production, would it work?

File: authors/settings.py [hunk 4]
```diff
@@ -152,4 +152,19 @@
...
+# DEFAULT_DOMAIN = 'https://www.yoursite.com:port'
```
Comment (FawazFarid) [inline authors/settings.py]: Do we need this comment?

**Review by FawazFarid — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/views.py (hunk 1 above)
Comment (ibetween) [reply authors/apps/authentication/views.py]: Cool. Will rectify this.

File: authors/settings.py (hunk 3 above)
Comment (ibetween) [reply authors/settings.py]: Not sure, let me use the official link from Lukorito to avoid surprises.

File: authors/settings.py (hunk 4 above)
Comment (ibetween) [reply authors/settings.py]: Noted. will get rid of it.

**Comment by lukorito** [comment]
Some great work in modifying the tests. This should be merged so that all incoming tests to follow suite @FawazFarid

**Review by FawazFarid — APPROVED** [review]

**Pull request merged**